lets-plot-compose/src/androidMain/kotlin/org/jetbrains/letsPlot/skia/compose/android/PlotViewContainer.kt

```diff
@@ -90,76 +103,94 @@ internal class PlotViewContainer(
+        val scaledSize = DoubleVector(w.toDouble(), h.toDouble()).mul(1.0 / density)
         val plotSize = PlotSizeUtil.preferredFigureSize(
             processedSpec,
             preserveAspectRatio!!,
-            containerSize
+            scaledSize
         )
-
-        val plotX = if (plotSize.x >= containerSize.x) 0.0 else {
-            (containerSize.x - plotSize.x) / 2
+        val plotX = if (plotSize.x >= scaledSize.x) 0.0 else {
+            (scaledSize.x - plotSize.x) / 2
         }
-        val plotY = if (plotSize.y >= containerSize.y) 0.0 else {
-            (containerSize.y - plotSize.y) / 2
+        val plotY = if (plotSize.y >= scaledSize.y) 0.0 else {
+            (scaledSize.y - plotSize.y) / 2
         }
-
         val bounds = DoubleRectangle(
             DoubleVector(plotX, plotY),
             plotSize
         )
 
-        val plotComponent = MonolithicSkiaAndroid.buildPlotFromProcessedSpecs(
-            ctx = context,
-            plotSize = plotSize,
-            plotSpec = processedSpec as MutableMap<String, Any>,
-        ) { messages ->
-            computationMessagesHandler(messages)
-        }
+        val unscaledOrigin = bounds.origin.mul(density.toDouble())
+        val unscaledSize = bounds.dimension.mul(density.toDouble())
+
+        val left = unscaledOrigin.x.toInt()
+        val top = unscaledOrigin.y.toInt()
+        val width = unscaledSize.x.toInt()
+        val height = unscaledSize.y.toInt()
+
+        if (!reusePlotSvgPanel) {
+            disposePlotView()
+
+            // https://stackoverflow.com/questions/25516363/how-to-properly-add-child-views-to-view-group
+            post {
+                check(childCount == 0) { "Can't revalidate: childCount = $childCount but should be 0." }
+
+                val plotComponent = MonolithicSkiaAndroid.buildPlotFromProcessedSpecs(
+                    ctx = context,
+                    plotSize = plotSize,
+                    plotSpec = processedSpec as MutableMap<String, Any>,
+                ) { messages ->
+                    computationMessagesHandler(messages)
+                }
+
+                val params = LayoutParams(width, height).also {
+                    it.leftMargin = left
+                    it.topMargin = top
+                }
+
+                addView(plotComponent, params)
+//            this.requestLayout()
+//            this.forceLayout()
+
+                // Add measure/layout plotView.
+                // Without this trick Skia layer do not re-draw after thw screen (device) rotation.
+                measureChild(
+                    plotComponent,
+                    MeasureSpec.makeMeasureSpec(width, MeasureSpec.EXACTLY),
+                    MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY)
+                )
+                plotComponent.layout(left, top, left + width, top + height)
+            }
+        } else {
+            post {
+                plotSvgRegistration.dispose()
```

Comment:
Can we have smth like `svgPanel.cleanAll()` here?

Reply:
I think `cleanAll()` doesn't work with `SvgPanel`. `plotSvgRegistration` is constructed by `Monolithic` only when `SvgPanel` is used to display a plot. We can also use `SvgPanel` to display a static SVG, in this case `cleanAll()` is not needed at all. More to say, `cleanAll()` needs a mutable property in `SvgPanel` to store the cleanup proc.

Reply:
> I think cleanAll() doesn't work with SvgPanel. plotSvgRegistration is constructed by Monolithic only when SvgPanel is used to display a plot.

Not sure what you mean, we need to add smth like `cleanAll()` to `SvgPanel`. As I can see, in 

`MonolithicSkiaAwt.buildPlotFromProcessedSpecs()` instead of 
```
val reg = CompositeRegistration()
```

we can have 

```
val reg = dest.contentRegistration
```

So that svg panel could cleanup its content later.